tools/lambda-promtail/lambda-promtail/eventbridge.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/go-kit/log"
+)
+
+// S3Detail encodes the message structure in EventBridge s3 notifications.
+// https://docs.aws.amazon.com/AmazonS3/latest/userguide/ev-events.html
+type S3Detail struct {
+	Version string `json:"version"`
+	Bucket  struct {
+		Name string `json:"name"`
+	} `json:"bucket"`
+	Object S3ObjectDetail `json:"object"`
+}
+
+type S3ObjectDetail struct {
+	Key       string `json:"key"`
+	Size      int    `json:"size"`
+	ETag      string `json:"etag"`
+	VersionID string `json:"version-id"`
+	Sequencer string `json:"sequencer"`
+}
+
+type s3EventProcessor func(ctx context.Context, ev *events.S3Event, pc Client, log *log.Logger) error
+
+func processEventBridgeEvent(ctx context.Context, ev *events.CloudWatchEvent, pc Client, log *log.Logger, process s3EventProcessor) error {
+	// lambda-promtail should only be used with S3 object creation events, since those indicate that a new file has been
+	// added to bucket, and need to be fetched and parsed accordingly.
+	if !(ev.Source == "aws.s3" && ev.DetailType == "Object Created") {
+		return fmt.Errorf("event bridge event type not supported")
+	}
+
+	var eventDetail S3Detail
+	if err := json.Unmarshal(ev.Detail, &eventDetail); err != nil {
+		return err
+	}
+
+	// TODO(thepalbi): how to fill bucket owner?
```

Comment:
the current logic extracts [this other field](https://github.com/grafana/loki/blob/4cd76919100fe59d41bd832bb07518f267d4ea1d/tools/lambda-promtail/lambda-promtail/s3.go#L208) from the original s3 events (the owner's principal ID), but this ends up not being used. See [here](https://github.com/grafana/loki/blob/4cd76919100fe59d41bd832bb07518f267d4ea1d/tools/lambda-promtail/lambda-promtail/s3.go#L248).

EventBridge events don't include the principal ID, but include the bucket AWS acct id, which is the value that should be used in [here](https://github.com/grafana/loki/blob/4cd76919100fe59d41bd832bb07518f267d4ea1d/tools/lambda-promtail/lambda-promtail/s3.go#L248):

See doc below in the field where it was supposed to be used in:
```
	// The account ID of the expected bucket owner. If the bucket is owned by a
	// different account, the request will fail with an HTTP 403 (Access Denied) error.
	ExpectedBucketOwner *string
```